google-cloud-firestore/src/main/java/com/google/cloud/firestore/FieldPath.java

```diff
@@ -83,6 +85,57 @@ static FieldPath fromDotSeparatedString(String field) {
     return empty().append(field);
   }
 
+  /** Creates a {@code FieldPath} from a server-encoded field path. */
```

Comment:
Add a comment that this is duplicated from the Android implementation and provide a link. (https://cs.opensource.google/firebase-sdk/firebase-android-sdk/+/master:firebase-firestore/src/main/java/com/google/firebase/firestore/model/FieldPath.java;l=47;drc=00d46269f1ef9ce0f1dd69ab727137a84332d85c)